packages/next/server/react-dom-server.ts

```diff
@@ -0,0 +1,13 @@
+let ReactDOMServer: typeof import('react-dom/server')
+
+if (process.browser) {
+  if (process.env.NODE_ENV === 'production') {
+    ReactDOMServer = require('react-dom/cjs/react-dom-server.browser.production.min')
+  } else {
+    ReactDOMServer = require('react-dom/cjs/react-dom-server.browser.development')
+  }
+} else {
+  ReactDOMServer = require('react-dom/server')
+}
+
+export default ReactDOMServer
```

Comment:
We should do this via aliases instead (#31035). Longer term the idea is replace those aliases with [conditional exports](https://nodejs.org/api/packages.html#conditional-exports).